[Build] compiling errors for armeabi-v7a: vfmaq_f32 undecared

### Describe the issue

I am building onnxruntime for armeabi-v7a, so arm in 32 bits, the older versions are fine but today (Sep 4 2026) I did a git pull and then I ran into compiling errors all of which are related to MLAS, in particular, vfmaq_f32 undeclared.

I think it is only for armv8 but I need to build for armv7.

### Urgency

_No response_

### Target platform

armv7 and android

### Build script

Below is how I build:
./build.sh \
  --config Release \
  --android \
  --android_abi armeabi-v7a \
  --android_api 29 \
  --android_ndk_path $ANDROID_NDK_PATH \
  --build_shared_lib \
  --parallel \
  --skip_tests \
  --use_nnapi \
  --use_xnnpack \
  --cmake_extra_defines \
    "onnxruntime_DISABLE_TRAINING=ON" \
    "onnxruntime_USE_NEON=ON" \
    "onnxruntime_DISABLE_TRAINING=ON" \
    "onnxruntime_ENABLE_CPU_FP32_NCHWc_OPS=ON" \
    "onnxruntime_DISABLE_EXCEPTIONS=ON" \
    "onnxruntime_DISABLE_RTTI=ON" \
    "onnxruntime_BUILD_UNIT_TESTS=OFF" \
    "onnxruntime_USE_MLAS=OFF" \  # doesn't help but kept anyway
    "MLAS_TARGET_ARCHITECTURE=generic" \  # doesn't help but kept anyway
    "CMAKE_CXX_FLAGS=-DMLAS_TARGET_ARM=1"  # doesn't help but kept anyway


### Error / output

In fact all errors are vfmaq_f32 undecared and from MLAS. For example:
[  6%] Building CXX object _deps/protobuf-build/CMakeFiles/libprotobuf-lite.dir/src/google/protobuf/generated_enum_util.cc.o
In file included from /home/weidu/WorkAtTPVision/test/xxx/onnxruntime/onnxruntime/core/mlas/lib/sgemm.cpp:18:
/home/weidu/WorkAtTPVision/test/xxx/onnxruntime/onnxruntime/core/mlas/lib/mlasi.h:2283:12: error: use of undeclared identifier 'vfmaq_f32'
    return vfmaq_f32(Vector3, Vector1, Vector2);
           ^


### Visual Studio Version

_No response_

### GCC / Compiler Version

_No response_

--- a/onnxruntime/core/mlas/lib/mlasi.h
+++ b/onnxruntime/core/mlas/lib/mlasi.h
@@ -2280,7 +2280,8 @@ MLAS_FLOAT32X4
 MlasMultiplyAddFloat32x4(MLAS_FLOAT32X4 Vector1, MLAS_FLOAT32X4 Vector2, MLAS_FLOAT32X4 Vector3)
 {
 #if defined(MLAS_NEON_INTRINSICS)
-#if defined(MLAS_TARGET_ARM)
+#if defined(__ANDROID__) && defined(MLAS_TARGET_ARM)
+    // Android armeabi-v7a ABI doesn't have vfmaq_f32()
     return vmlaq_f32(Vector3, Vector1, Vector2);
 #else
     return vfmaq_f32(Vector3, Vector1, Vector2);

